### DISCUSSION TITLE
Can't use copilot in Visual Studio with Error on ghost text request: (FetchError) tunneling socket could not be established, statusCode=

### DISCUSSION THREAD

**Discussion by qilincai** [community/community#69988] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Copilot can work normally in pycharm and vscode, but only Visual Studio cannot work normally.

Microsoft Visual Studio Community 2022 17.7.5

GitHub Copilot 1.116.0.0 (v1.116.0.0@5a26aa711)
GitHub Copilot Agent v1.116.0
[DEBUG] [agent] [2023-10-13T05:57:32.461Z] Agent service starting
[DEBUG] [agent] [2023-10-13T05:57:32.609Z] Telemetry initialized
[DEBUG] [proxy-socket-factory] [2023-10-13T05:57:32.918Z] Attempting to establish connection to proxy
[DEBUG] [proxy-socket-factory] [2023-10-13T05:57:33.050Z] Socket Connect returned status code,200
[DEBUG] [proxy-socket-factory] [2023-10-13T05:57:33.050Z] Successfully established tunneling connection to proxy
Auth Status: OK
Successfully authorized GitHub Copilot.
[DEBUG] [proxy-socket-factory] [2023-10-13T05:57:47.634Z] Attempting to establish connection to proxy
[DEBUG] [proxy-socket-factory] [2023-10-13T05:57:47.684Z] Socket Connect returned status code,403
[DEBUG] [getCompletions] [2023-10-13T06:01:29.098Z] Requesting completion at position 820:0, between "        string _sFilePath = \"\";\r\n" and "\r\n\r\n".
[INFO] [default] [2023-10-13T06:01:29.741Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[DEBUG] [proxy-socket-factory] [2023-10-13T06:01:29.743Z] Attempting to establish connection to proxy
[DEBUG] [proxy-socket-factory] [2023-10-13T06:01:29.765Z] Socket Connect returned status code,403
statusNotification: Warning: tunneling socket could not be established, statusCode=403
[ERROR] [ghostText] [2023-10-13T06:01:29.768Z] Error on ghost text request: (FetchError) tunneling socket could not be established, statusCode=403
[INFO] [RequestProposalsAsync] 0 completions available.
[DEBUG] [proxy-socket-factory] [2023-10-13T06:01:44.675Z] Attempting to establish connection to proxy
[DEBUG] [proxy-socket-factory] [2023-10-13T06:01:44.691Z] Socket Connect returned status code,403

**Comment by Akash1134** [comment]
Hi @qilincai ,

The error message "tunneling socket could not be established, statusCode=403" suggests that there's an issue with the network connection between your machine and the GitHub Copilot. 

We have a troubleshooting guide to help you get such issues sorted. 
- [Troubleshooting Network errors](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot)
- [Troubleshooting Firewall settings](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-firewall-settings-for-github-copilot)

Let us know how it goes. 

Thanks!

**Comment by psziom** [comment, marked as answer]
After months of thinking about this issue, today I have finally resolved it :)
I needed to turn off the automatic proxy settings in windows networking.
![turnOffProxySettings](https://github.com/community/community/assets/68161836/c5782fbc-8514-422b-bfd5-19b110486214)
